### PR TITLE
(feat) Order summary/history

### PR DESCRIPTION
**Pull request by pirupius**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary
This PR introduces order history nav in the patient chart 
- [x] Add `Orders` to patient chart side nav
- [x] Add list of all orders
- [x] Add filter for orders by order type
- [x] Add feature flag

## Screenshots

Screenshot 2023-11-28 at 17 54 59

https://github.com/openmrs/openmrs-esm-patient-chart/assets/15266028/70f051eb-17f4-47d1-8919-c4ccd25bd751

## Related Issue
[O3-2355](https://issues.openmrs.org/browse/O3-2355)

### CODE REVIEW

File: packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.tsx [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import React, { useCallback, useState } from 'react';
```
Comment (pirupius) [inline packages/esm-patient-labs-app/src/lab-orders/add-lab-order/lab-order-form.tsx:1]: This file was renamed in remote so this should be deleted

File: packages/esm-patient-labs-app/src/lab-orders/api.ts [hunk 2]
```diff
@@ -33,6 +33,8 @@ export function usePatientLabOrders(patientUuid: string, status: 'ACTIVE' | 'any
     [data],
   );
 
+  console.log('labOrders', labOrders);
```
Comment (pirupius) [inline packages/esm-patient-labs-app/src/lab-orders/api.ts:36]: delete

**Comment by gracepotma** [comment]
Maybe we can feature flag this until it's ready, but keep merging things in? What do you think @denniskigen ?

**Comment by brandones** [comment]
@pirupius I support @gracepotma 's idea. It would be nice to start merging things in and not have to review one giant PR all at once. Could you please put this behind a feature flag and get it ready for review as a WIP feature? Including, of course, plenty of screenshots or a good video tour of the feature.

**Comment by pirupius** [comment]
Sure thing @brandones will push the changes and update the description

File: packages/esm-patient-appointments-app/translations/km.json [hunk 3]
```diff
@@ -25,14 +25,14 @@
...
-  "daysOfWeek": "ថ្ងៃ​នៃ​ស​ប្តា​ហ៍",
+  "daysOfWeek": "ថ្ងៃ\u200bនៃ\u200bស\u200bប្តា\u200bហ៍",
   "discard": "បោះបង់",
   "durationInMinutes": "រយៈពេល (នាទី)",
   "edit": "កែសម្រួល",
   "editAppointment": "កែសម្រួលការណាត់ជួប",
   "endDate": "កាលបរិច្ឆេទបញ្ចប់",
   "invalidNumber": "លេខមិនត្រឹមត្រូវទេ",
-  "isRecurringAppointment": "តើ​នេះ​ជា​ការ​ណាត់​ជួប​ច្រើនដងមែនទេ?",
+  "isRecurringAppointment": "តើ\u200bនេះ\u200bជា\u200bការ\u200bណាត់\u200bជួប\u200bច្រើនដងមែនទេ?",
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply packages/esm-patient-appointments-app/translations/km.json:35]: Please revert the change in `km.json`

File: package.json [hunk 4]
```diff
@@ -5,7 +5,7 @@
     "packages/*"
   ],
   "scripts": {
-    "start": "openmrs develop --sources packages/esm-patient-chart-app/",
+    "start": "openmrs develop --sources packages/esm-patient-orders-app/",
```
[the remark this answers is not in the document]
Comment (vasharma05) [reply package.json:8]: Please undo this change.

File: packages/esm-patient-banner-app/translations/km.json [hunk 5]
```diff
@@ -2,8 +2,8 @@
   "actions": "ជ្រើសរើសមុខងារ",
   "activeVisit": "ការមកពិនិត្យជំងឺសកម្ម",
   "address": "អាស័យដ្ឋាន",
-  "address1": "អាសយដ្ឋាន​ជួរទី១",
-  "address2": "អ​ស​យ​ដ្ឋាន​ជួរ​ទី 2",
+  "address1": "អាសយដ្ឋាន\u200bជួរទី១",
```
Comment (vasharma05) [inline packages/esm-patient-banner-app/translations/km.json:5]: Please revert the km.json file

File: packages/esm-patient-immunizations-app/translations/km.json [hunk 6]
```diff
@@ -11,7 +11,7 @@
   "pleaseSelect": "Please select",
   "recentVaccination": "Recent vaccination",
   "save": "Save",
-  "seeAll": "ឃើញ​ទាំងអស់",
+  "seeAll": "ឃើញ\u200bទាំងអស់",
```
Comment (vasharma05) [inline packages/esm-patient-immunizations-app/translations/km.json:14]: Please revert

File: packages/esm-patient-labs-app/src/lab-orders/api.ts [hunk 7]
```diff
@@ -13,7 +13,7 @@ export const careSettingUuid = '6f0c9a92-6f24-11e3-af88-005056821db0';
  */
 export function usePatientLabOrders(patientUuid: string, status: 'ACTIVE' | 'any') {
   const { labOrderTypeUuid: labOrderTypeUUID } = (useConfig() as ConfigObject).orders;
-  const ordersUrl = `/ws/rest/v1/order?patient=${patientUuid}&careSetting=${careSettingUuid}&status=${status}&orderType=${labOrderTypeUUID}`;
+  const ordersUrl = `/ws/rest/v1/order?v=full&patient=${patientUuid}&careSetting=${careSettingUuid}&status=${status}&orderType=${labOrderTypeUUID}`;
```
Comment (vasharma05) [inline packages/esm-patient-labs-app/src/lab-orders/api.ts:16]: `v=full` can really bring a lot of changes. 
Please review the fields actually required and pass the relevant fields in the `v`
Thanks!

File: packages/esm-patient-orders-app/src/api/api.ts [hunk 8]
```diff
@@ -135,3 +142,48 @@ export function useOrderEncounter(patientUuid: string): {
   }, [isLoadingSystemVisitSetting, errorFetchingSystemVisitSetting, visit, todayEncounter, systemVisitEnabled]);
   return results;
 }
+
+export function useOrderTypes() {
+  const orderTypesUrl = `/ws/rest/v1/ordertype`;
+  const { data, error, isLoading, isValidating } = useSWR<FetchResponse, Error>(orderTypesUrl, openmrsFetch);
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/api/api.ts:148]: What is the type of OrderType, can you please add the relevant type?
Thanks!

File: packages/esm-patient-orders-app/src/api/api.ts [hunk 9]
```diff
@@ -135,3 +142,48 @@ export function useOrderEncounter(patientUuid: string): {
...
+
+  return {
+    data: data?.data?.results,
+    error,
+    isLoading,
+    isValidating,
+  };
+}
+
+export function usePatientOrders(patientUuid: string, status: 'ACTIVE' | 'any', orderType?: string) {
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/api/api.ts:158]: This is duplicated in [packages/esm-patient-labs-app/src/lab-orders/api.ts](https://github.com/openmrs/openmrs-esm-patient-chart/pull/1463/files#diff-d2930c7ae1ee419fd3e61a7985f5aca6c72931f1addb5d3fd0129d53f7e76395). I believe we can move this to `patient-common-lib`.
Thanks!

File: packages/esm-patient-orders-app/src/api/api.ts [hunk 10]
```diff
@@ -135,3 +142,48 @@ export function useOrderEncounter(patientUuid: string): {
...
+
+  const orders = useMemo(
+    () =>
+      data?.data?.results
+        ? data.data.results?.sort((order1, order2) => (order2.dateActivated > order1.dateActivated ? 1 : -1))
+        : null,
+    [data],
+  );
+
+  console.log('orders', orders);
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/api/api.ts:180]: Please remove `console` statement

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 11]
```diff
@@ -0,0 +1,455 @@
...
+      isSortable: true,
+    },
+    {
+      key: 'priority',
+      header: t('priority', 'Priority'),
+      isSortable: true,
+    },
+    {
+      key: 'orderedBy',
+      header: t('orderedBy', 'Ordered By'),
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:107]:
```suggestion
      header: t('orderedBy', 'Ordered by'),
```

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 12]
```diff
@@ -0,0 +1,455 @@
...
+      gender: getGender(patient?.patient?.gender),
+      location: patient?.patient?.address?.[0].city,
+      identifiers: identifiers?.length ? identifiers.map(({ value, type }) => value) : [],
+    };
+  }, [patient, t, excludePatientIdentifierCodeTypes?.uuids]);
+
+  // trigger a refetch to parent component if selectedOrderType changes
+  useEffect(() => {
+    // TODO: this is a hack to trigger a refetch of the parent component
+  }, [selectedOrderTypeUuid]);
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:201]: Can't we have the `selectedOrderTypeUuid` state be created in the parent itself?
Thanks!

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 13]
```diff
@@ -0,0 +1,455 @@
...
+
+  const onBeforeGetContentResolve = useRef(null);
+
+  useEffect(() => {
+    if (isPrinting && onBeforeGetContentResolve.current) {
+      onBeforeGetContentResolve.current();
+    }
+  }, [isPrinting]);
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:209]: Missing dependency: `onBeforeGetContentResolv`

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 14]
```diff
@@ -0,0 +1,455 @@
...
+      setIsPrinting(false);
+    },
+  });
+
+  return (
+    <>
+      {(() => {
+        if (isLoading) return <DataTableSkeleton role="progressbar" compact={!isTablet} zebra />;
+        if (isError) return <ErrorState error={isError} headerTitle={title} />;
+        if (tableRows) {
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:232]: Can we please have these outside of the return as separate `if() return <...>`, it increases the code readablity,
Thanks!

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 15]
```diff
@@ -0,0 +1,455 @@
...
+    // openOrderForm({ order: orderItem });
+  }, [orderItem, openOrderForm]);
+
+  const handleAddResultsClick = useCallback(() => {
+    // openOrderForm({ order: orderItem, addResults: true });
+  }, [orderItem, openOrderForm]);
+
+  const handleCancelClick = useCallback(() => {
+    // setItems(items.filter((x) => x.uuid !== orderItem.uuid));
+  }, [items, orderItem, setItems]);
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:400]: Commented `useCallbacks`

File: packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx [hunk 16]
```diff
@@ -0,0 +1,455 @@
...
+  } else if (y == undefined) {
+    return 1;
+  } else if (x < y) {
+    return -1;
+  } else if (x > y) {
+    return 1;
+  } else {
+    return 0;
+  }
+}
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/components/orders-details-table.component.tsx:453]: We can move these in `esm-patient-common-lib`

File: packages/esm-patient-orders-app/src/index.ts [hunk 17]
```diff
@@ -25,3 +27,13 @@ registerWorkspace({
 });
 
 export const orderBasketActionMenu = getSyncLifecycle(orderBasketActionMenuComponent, options);
+
+export const ordersDashboardLink = getSyncLifecycle(
+  createDashboardLink({
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/index.ts:32]:
```suggestion
	// t('Orders', 'Orders')
  createDashboardLink({
```

For support of translating the orders text

File: packages/esm-patient-orders-app/src/orders-summary/orders-summary.component.tsx [hunk 18]
```diff
@@ -0,0 +1,25 @@
...
+import { useTranslation } from 'react-i18next';
+import OrderDetailsTable from '../components/orders-details-table.component';
+
+export interface OrdersSummaryProps {
+  patientUuid: string;
+}
+
+const OrdersSummary: React.FC<OrdersSummaryProps> = ({ patientUuid }) => {
+  const { t } = useTranslation();
+  const ordersDisplayText = t('Orders', 'Orders');
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/orders-summary/orders-summary.component.tsx:11]:
```suggestion
  const ordersDisplayText = t('orders', 'Orders');
```

File: packages/esm-patient-orders-app/src/print/print.component.tsx [hunk 19]
```diff
@@ -0,0 +1,63 @@
...
+            />
+          </svg>
```
Comment (vasharma05) [inline packages/esm-patient-orders-app/src/print/print.component.tsx:38]: This should be configurable, since other orgs might want to add their logo

**Review by vasharma05 — CHANGES_REQUESTED** [review]

File: packages/esm-patient-orders-app/src/api/api.ts (hunk 9 above)
Comment (pirupius) [reply packages/esm-patient-orders-app/src/api/api.ts:158]: Agreed. Let me move it

**Comment by pirupius** [comment]
@denniskigen @vasharma05 if this is all good, can we please get this in. It's been long overdue.

**Pull request merged**